### ISSUE TITLE
Using --abort-on-error flag causes an error when YouTube channel has no videos uploaded

### ISSUE THREAD

**Issue by PAI5REECHO** [yt-dlp/yt-dlp#5906]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

When attempting to download a YouTube channel with no uploads while using `--abort-on-error`, `yt-dlp` exits with an error, which is unexpected (one would reasonably expect it to not cause an error and just not download anything) and inconsistent behavior (when attempting to download a Rumble channel with no videos no such error occurs). At most I would expect a warning message, but certainly not a fatal error.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```console
$ yt-dlp -vU --ignore-config --abort-on-error 'https://youtube.com/@cats'
[debug] Command-line config: ['-vU', '--ignore-config', '--abort-on-error', 'https://youtube.com/@cats']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.11.1 (CPython x86_64 64bit) - Linux-6.0.2-x86_64-with (OpenSSL 3.0.7 1 Nov 2022)
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [youtube:tab] Extracting URL: https://youtube.com/@cats
[youtube:tab] @cats: Downloading webpage
[debug] [youtube:tab] Selected tab: 'featured' (home), Requested tab: ''
[youtube:tab] Downloading all uploads of the channel. To download only the videos in a specific tab, pass the tab's URL
ERROR: [youtube:tab] @cats: This channel has no uploads
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 674, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/extractor/youtube.py", line 4285, in wrapper
    info_dict = func(self, url, smuggled_data)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/extractor/youtube.py", line 6029, in _real_extract
    raise ExtractorError('This channel has no uploads', expected=True)

$ yt-dlp -vU --ignore-config --abort-on-error 'https://rumble.com/abcd'
[debug] Command-line config: ['-vU', '--ignore-config', '--abort-on-error', 'https://rumble.com/abcd']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.11.1 (CPython x86_64 64bit) - Linux-6.0.2-x86_64-with (OpenSSL 3.0.7 1 Nov 2022)
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://rumble.com/abcd
[generic] abcd: Downloading webpage
[redirect] Following redirect to https://rumble.com/c/c-114491
[debug] [RumbleChannel] Extracting URL: https://rumble.com/c/c-114491
[download] Downloading playlist: c-114491
[RumbleChannel] c-114491: Downloading page 1
[RumbleChannel] c-114491: Downloading page 2
[RumbleChannel] Playlist c-114491: Downloading 0 videos of 0
[download] Finished downloading playlist: c-114491
```

**Comment by pukkandan** [comment]
@coletdjnz Do you remember what was the reason we decided on error?

**Comment by coletdjnz** [comment]
> @coletdjnz Do you remember what was the reason we decided on error?

as the tabs do not exist.

I recall discussing the possibility of returning an empty playlist in this case.

**Comment by AYehia0** [comment]
I think this issue should be closed as of today, yt-dlp does throw this error : `ERROR: [youtube:tab] @<channel>: This channel has no uploads`

```
~/x ❯ yt-dlp -vU --ignore-config --abort-on-error 'https://youtube.com/@cats'                                                              ⬢
[debug] Command-line config: ['-vU', '--ignore-config', '--abort-on-error', 'https://youtube.com/@cats']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34] (zip)
[debug] Python 3.11.6 (CPython x86_64 64bit) - Linux-6.6.7-zen1-1-zen-x86_64-with-glibc2.38 (OpenSSL 3.2.0 23 Nov 2023, glibc 2.38)
[debug] exe versions: ffmpeg 6.1 (setts), ffprobe 6.1, rtmpdump 2.4
[debug] Optional libraries: certifi-2023.11.17, requests-2.23.0, secretstorage-3.3.3, sqlite3-3.44.2, urllib3-1.25.11, websockets-11.0.3
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1798 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.12.30 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.12.30 from yt-dlp/yt-dlp)
[youtube:tab] Extracting URL: https://youtube.com/@cats
[youtube:tab] @cats: Downloading webpage
[debug] [youtube:tab] Selected tab: 'featured' (home), Requested tab: ''
[youtube:tab] Downloading all uploads of the channel. To download only the videos in a specific tab, pass the tab's URL
[download] Downloading playlist: Cats
[youtube:tab] Playlist Cats: Downloading 0 items
[download] Finished downloading playlist: Cats
```

**Comment by AYehia0** [comment]
@bashonly sorry I meant to say that this issue has been solved, as I don't see that error

**Comment by pukkandan** [comment]
Was fixed in 044886c220620a7679109e92352890e18b6079e3. Forgot to close issue
